Installation issue: aocl-utils

### Steps to reproduce the issue

```console
$ spack env create aocl-utils
$ spack env activate -p aocl-utils
$ spack install --add aocl-utils

==> Concretized 1 spec
 -   zduojyn  aocl-utils@5.0~doc~examples~ipo+shared~tests build_system=cmake build_type=Release generator=make arch=linux-ubuntu24.04-zen4
 -   lfgksmq      ^cmake@3.31.6~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=linux-ubuntu24.04-zen4
 -   n6eshm4          ^curl@8.11.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl arch=linux-ubuntu24.04-zen4
 -   eblxinv              ^nghttp2@1.65.0 build_system=autotools arch=linux-ubuntu24.04-zen4
 -   46ayiqp                  ^diffutils@3.10 build_system=autotools arch=linux-ubuntu24.04-zen4
 -   bwbnig2                      ^libiconv@1.17 build_system=autotools libs:=shared,static arch=linux-ubuntu24.04-zen4
 -   d2vpwpm              ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu24.04-zen4
 -   wygtrii                  ^ca-certificates-mozilla@2025-02-25 build_system=generic arch=linux-ubuntu24.04-zen4
 -   fczb6pp                  ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-ubuntu24.04-zen4
 -   wdohljk                      ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=linux-ubuntu24.04-zen4
 -   m5lqart                      ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-ubuntu24.04-zen4
 -   iifuhti                      ^gdbm@1.23 build_system=autotools arch=linux-ubuntu24.04-zen4
 -   tmt6f6m                          ^readline@8.2 build_system=autotools patches:=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=linux-ubuntu24.04-zen4
 -   ox6uwu3              ^pkgconf@2.3.0 build_system=autotools arch=linux-ubuntu24.04-zen4
 -   ny3a2k6          ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-ubuntu24.04-zen4
 -   prkon7f          ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu24.04-zen4
 -   l3mwbjo      ^compiler-wrapper@1.0 build_system=generic arch=linux-ubuntu24.04-zen4
[e]  yx4a6ss      ^gcc@13.3.0~binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' arch=linux-ubuntu24.04-zen4
 -   ef3jn5w      ^gcc-runtime@13.3.0 build_system=generic arch=linux-ubuntu24.04-zen4
[e]  jalaln3      ^glibc@2.39 build_system=autotools arch=linux-ubuntu24.04-zen4
 -   3xsbbua      ^gmake@4.4.1~guile build_system=generic arch=linux-ubuntu24.04-zen4

<installation of the first 20 packages snipped>

==> Installing aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3ie [21/21]
==> No binary for aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3ie found: installing from source
==> Using cached archive: /home/ali/software/spack/var/spack/cache/_source-cache/archive/ee/ee2e5d47f33a3f673b3b6fcb88a7ef1a28648f407485ad07b6e9bf1b86159c59.tar.gz
==> No patches needed for aocl-utils
==> aocl-utils: Executing phase: 'cmake'
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Error: ProcessError: Command exited with status 1:
    '/home/ali/software/spack/opt/spack/linux-zen4/cmake-3.31.6-lfgksmqzsw4sbyja3f7qrt6j2dd2mc2x/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/home/ali/software/spack/opt/spack/linux-zen4/aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3ie' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/home/ali/software/spack/opt/spack/linux-zen4/aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3ie/lib;/home/ali/software/spack/opt/spack/linux-zen4/aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3ie/lib64' '-DCMAKE_PREFIX_PATH:STRING=/home/ali/software/spack/opt/spack/linux-zen4/cmake-3.31.6-lfgksmqzsw4sbyja3f7qrt6j2dd2mc2x;/home/ali/software/spack/opt/spack/linux-zen4/compiler-wrapper-1.0-l3mwbjobl2pzuy3d3cqlamkjbqkwqutr;/home/ali/software/spack/opt/spack/linux-zen4/gcc-runtime-13.3.0-ef3jn5watkvb223czqb2qn5htunv2uco;/home/ali/software/spack/opt/spack/linux-zen4/gmake-4.4.1-3xsbbuasxbygjk5kq4ekwds54mfmqrdh' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DBUILD_SHARED_LIBS:BOOL=ON' '-DCMAKE_INSTALL_LIBDIR:STRING=lib' '-DAU_BUILD_DOCS:BOOL=OFF' '-DAU_BUILD_TESTS:BOOL=OFF' '-DAU_BUILD_EXAMPLES:BOOL=OFF' '/tmp/ali/spack-stage/spack-stage-aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3ie/spack-src'

3 errors found in build log:
     11    -- Detecting CXX compile features
     12    -- Detecting CXX compile features - done
     13    -- Detecting C compiler ABI info
     14    -- Detecting C compiler ABI info - failed
     15    -- Check for working C compiler: /home/ali/software/spack/opt/spack/linux-zen4/compiler-wrapper-1.0-l3mwbjobl2pzuy3d3cqlamkjbqkwqutr/libexec/spack/gcc/gcc
     16    -- Check for working C compiler: /home/ali/software/spack/opt/spack/linux-zen4/compiler-wrapper-1.0-l3mwbjobl2pzuy3d3cqlamkjbqkwqutr/libexec/spack/gcc/gcc - broken
  >> 17    CMake Error at /home/ali/software/spack/opt/spack/linux-zen4/cmake-3.31.6-lfgksmqzsw4sbyja3f7qrt6j2dd2mc2x/share/cmake-3.31/Modules/CMakeTestCCompiler.cmake:67 (message):
     18      The C compiler
     19
     20        "/home/ali/software/spack/opt/spack/linux-zen4/compiler-wrapper-1.0-l3mwbjobl2pzuy3d3cqlamkjbqkwqutr/libexec/spack/gcc/gcc"
     21
     22      is not able to compile a simple test program.
     23

     ...

     28        Run Build Command(s): /home/ali/software/spack/opt/spack/linux-zen4/cmake-3.31.6-lfgksmqzsw4sbyja3f7qrt6j2dd2mc2x/bin/cmake -E env VERBOSE=1 /home/ali/software/spack/opt/spack/linux-zen4/gmake-4.4.1-3xsbbuasxbygjk5kq4ekwds54mfmqrdh/bin/gmake -f Makefile
           cmTC_ce6ca/fast
     29        /home/ali/software/spack/opt/spack/linux-zen4/gmake-4.4.1-3xsbbuasxbygjk5kq4ekwds54mfmqrdh/bin/gmake  -f CMakeFiles/cmTC_ce6ca.dir/build.make CMakeFiles/cmTC_ce6ca.dir/build
     30        gmake[1]: Entering directory '/tmp/ali/spack-stage/spack-stage-aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3ie/spack-build-zduojyn/CMakeFiles/CMakeScratch/TryCompile-Fz4BU5'
     31        Building C object CMakeFiles/cmTC_ce6ca.dir/testCCompiler.c.o
     32        /home/ali/software/spack/opt/spack/linux-zen4/compiler-wrapper-1.0-l3mwbjobl2pzuy3d3cqlamkjbqkwqutr/libexec/spack/gcc/gcc    -o CMakeFiles/cmTC_ce6ca.dir/testCCompiler.c.o -c /tmp/ali/spack-stage/spack-stage-aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3i
           e/spack-build-zduojyn/CMakeFiles/CMakeScratch/TryCompile-Fz4BU5/testCCompiler.c
     33        /home/ali/software/spack/opt/spack/linux-zen4/compiler-wrapper-1.0-l3mwbjobl2pzuy3d3cqlamkjbqkwqutr/libexec/spack/gcc/gcc: 1: eval: SPACK_CC_LINKER_ARG: ERROR: LINKER ARG WAS NOT SET, MAYBE THE PACKAGE DOES NOT DEPEND ON CC?
  >> 34    gmake[1]: *** [CMakeFiles/cmTC_ce6ca.dir/build.make:81: CMakeFiles/cmTC_ce6ca.dir/testCCompiler.c.o] Error 2
     35        gmake[1]: Leaving directory '/tmp/ali/spack-stage/spack-stage-aocl-utils-5.0-zduojyn3db7h3sfvmwzdwzyhncgkb3ie/spack-build-zduojyn/CMakeFiles/CMakeScratch/TryCompile-Fz4BU5'
  >> 36        gmake: *** [Makefile:134: cmTC_ce6ca/fast] Error 2
     37
     38
     39
     40
     41
     42      CMake will not be able to correctly generate this project.
</pre></details>


### Information on your system

```console
* **Spack:** 1.0.0.dev0 (3da44cff0bb1bce839bd56e560a9d0d35d12b727)
* **Python:** 3.12.3
* **Platform:** linux-ubuntu24.04-zen4
```

### Additional information

[spack-build-out.txt](https://github.com/user-attachments/files/19708120/spack-build-out.txt)
[spack-build-env.txt](https://github.com/user-attachments/files/19708119/spack-build-env.txt)

Pinging @amd-toolchain-support as instructed. Thanks for the help y'all.

I encountered this when installing `amd-aocl`, which failed due to this. I hope this is the only problem and the rest will complete successfully after we figure this out?

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/aocl-da/package.py
+++ b/var/spack/repos/builtin/packages/aocl-da/package.py
@@ -53,6 +53,8 @@ class AoclDa(CMakePackage):
     )
 
     depends_on("c", type="build")
+    depends_on("cxx", type="build")
+    depends_on("fortran", type="build")
 
     depends_on("cmake@3.22:", type="build")
     for vers in ["5.0"]:

--- a/var/spack/repos/builtin/packages/aocl-utils/package.py
+++ b/var/spack/repos/builtin/packages/aocl-utils/package.py
@@ -47,7 +47,8 @@ class AoclUtils(CMakePackage):
     variant("shared", default=True, when="@4.2:", description="build shared library")
     variant("examples", default=False, description="enable examples")
 
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
 
     depends_on("cmake@3.22:", type="build")
     depends_on("doxygen", when="+doc")

